### ISSUE TITLE
[Core] /api/cluster_status treats placement groups differently than ray status

### ISSUE THREAD

**Issue by alanwguo** [ray-project/ray#24309]
### What happened + What you expected to happen

When launching a simple cluster with placement groups that reserves CPUs, the output of `ray status` CLI command and the `/api/cluster_status` API endpoint treats reserved CPUs differently.

api/cluster_status: https://gist.github.com/alanwguo/a2349057b5618d72ba1d4c8a66df57ab
Excerpt: 
```
ResourceUsage: 2.0/8.0 CPU
```

ray status:
```
======== Autoscaler status: 2022-04-28 11:45:37.854142 ========
Node status
---------------------------------------------------------------
Healthy:
 1 head-node-type
Pending:
 (no pending nodes)
Recent failures:
 (no failures)

Resources
---------------------------------------------------------------
Usage:
 0.0/8.0 CPU (0.0 used of 2.0 reserved in placement groups)
 0.00/18.209 GiB memory
 0.00/9.104 GiB object_store_memory

Demands:
 (no resource demands)
```

Notice the difference between CPUs "used" and CPUs "reserved".

### Versions / Dependencies

ray 1.12.0
python 3.9
ubuntu

### Reproduction script

```
from ray.util.placement_group import (
  placement_group,
  placement_group_table,
  remove_placement_group
)
import ray

bundle1 = {"CPU": 2}
pg = placement_group([bundle1], strategy="STRICT_PACK")

ray.get(pg.ready())
```

### Issue Severity

Low: It annoys or frustrates me.

**Comment by xwjiang2010** [comment]
I experienced the same thing recently as well.

**Comment by rkooo567** [comment]
Hmm it is because the ray status parses the data generated from `cluster_status` and makes it more user-friendly format like the one in `ray status`. I think since the processing is done on the client side (ray status CLI command), the same information is not available to cluster_status API. 

"Reserved by placement group" means it allocates 2 cpus and create 2 pg custom resources (that look like `group_[id]_[index]_blah`). 

@alanwguo in the cluster status API output, do you see resources that look like ^?

**Comment by alanwguo** [comment]
Yes, in the response: https://gist.github.com/alanwguo/a2349057b5618d72ba1d4c8a66df57ab
I likely receive enough information to convert it into a more user-friendly format.

Can we move that logic into the API? Otherwise the product will need to re-implement the logic and it doesn't make sense to maintain it in two places.

**Comment by rkooo567** [comment]
SGTM. What's the reasonable deadline for this task?

**Comment by richardliaw** [comment]
Can we do it by 6/22? So in 2 weeks, @rkooo567 ?

**Comment by richardliaw** [comment]
Labeling this as 2.0 release blocker for now; feel free to negotiate.

**Comment by rkooo567** [comment]
Any reason why this needs to be a 2.0 blocker? I believe the endpoint is not public (so it doesn't have to be done before Ray 2.0). cc @alanwguo thoughts?

**Comment by richardliaw** [comment]
@rkooo567 recommend that you follow up offline, since this is affecting some AS users.

**Comment by rkooo567** [comment]
Let me handle this in the next sprint. I totally missed the issue.

**Comment by rkooo567** [comment]
@alanwguo set this as P2 for now. Please lmk if we should bump up the priority.

**Comment by cszhu** [comment]
Hello! This P3 issue has seen no activity in at least a year. It will be closed in 2 weeks as part of ongoing cleanup efforts.

Please remove the `pending-cleanup` label if you believe this issue should remain open.

Thanks for contributing to Ray!
